Generator/GLibGenerator.cs

```diff
@@ -6,6 +7,14 @@ namespace Generator
 {
     public class GLibGenerator : Generator<GLibTemplateLoader>
     {
+        enum StructType
```

Comment:
Really, the entire thing should probably go in `Generator.cs`, since GLib will never need to generate type classes anyway (GObject-only feature). I placed it in GLibGenerator to start with, since that's all we're generating at the moment. I'll go ahead and put `StructType` + `GetStructType` in the base class like you suggested, and we can look at sharing more code down the line.

It's not quite ready to be merged as TypeResolver is designed for writing argument syntax as opposed to field syntax (e.g. `ref PtrArray` is acceptable in a method declaration but would need to be `IntPtr` in a struct).